charms/reactive/relations.py

```diff
@@ -200,10 +203,14 @@ def _find_impl(cls, role, interface):
         """
         # The module has already been discovered and imported.
         module = 'relations.{}.{}'.format(interface, role)
-        if module in sys.modules:
-            return cls._find_subclass(sys.modules[module])
-        else:
-            return None
+        if module not in sys.modules:
+            try:
+                _append_path(hookenv.charm_dir())
+                _append_path(os.path.join(hookenv.charm_dir(), 'hooks'))
```

Comment:
Yeah, I had the same thought, that maybe I should factor it into a helper, but then it's only used in two places.  *shrug*